Fixes #190 - Moving from Jade templating language to Pug

willsteinmetz opened the pull request:
This pull request provides the transition from Jade to Pug. One change that will need to be made in the future is when the Pug 2.0.0 version comes out of beta.

petecoop commented:
This looks good, only issue I see here is that it's using a beta package. I'd rather use stable if we can

willsteinmetz commented:
There currently aren't any stable Pug packages yet. There are two super old versions and then it jumps to the 2.0.0 alphas and betas. That's the only reason I went with the beta. Hopefully it comes out of beta soon.

petecoop commented:
Okay I'd like to wait on the stable release then before merging really.

willsteinmetz commented:
Makes sense. I'll keep an eye out and update the PR accordingly.

willsteinmetz commented:
@petecoop After a very, very, very long time, this is hopefully good to go now. Please let me know if I need to make any other changes.

petecoop commented on README.md:
```diff
@@ -13,7 +13,7 @@ An Expressjs generator for Yeoman, based on the express command line tool.
 - .editorconfig for consistent coding styles within text editors
 - Support View engines:
-  - [Jade](http://jade-lang.com/)
+  - [Pug](http://jade-lang.com/)
```
https://pugjs.org

petecoop commented on .dependencies.json:
```diff
@@ -4,7 +4,7 @@
   "cookie-parser": "^1.3.3",
   "body-parser": "^1.13.3",
-  "jade": "^1.11.0",
+  "pug": "^2.0.0-beta5",
```
Lets use the same version everywhere!

petecoop commented on app/templates/basic-shared/package.json:
```diff
@@ -15,8 +15,8 @@
-    "body-parser": "^1.13.3"<% if(options.viewEngine == 'jade'){ %>,
-    "jade": "^1.11.0"<% } %><% if(options.viewEngine == 'ejs'){ %>,
+    "body-parser": "^1.13.3"<% if(options.viewEngine == 'pug'){ %>,
+    "pug": "^2.0.0-beta5"<% } %><% if(options.viewEngine == 'ejs'){ %>,
```
version!

willsteinmetz commented:
Argh, I'll push an update in a few minutes.

![Disappointed](https://media.giphy.com/media/3oAt21Fnr4i54uK8vK/giphy.gif)

willsteinmetz commented:
Alrighty @petecoop, I updated the noted files and actually found one other that still hadn't been updated. I did a search of the project for Jade and found no more instances so hopefully we've squashed them all now.

petecoop commented:
LGTM!
